drivers/timers/Kconfig

```diff
@@ -366,13 +366,21 @@ if WATCHDOG_AUTOMONITOR
 
 choice
 	prompt "Auto-monitor keepalive by"
-	default WATCHDOG_AUTOMONITOR_BY_TIMER
+	default WATCHDOG_AUTOMONITOR_BY_WDOG
 
 config WATCHDOG_AUTOMONITOR_BY_CAPTURE
 	bool "Capture callback"
 
+config WATCHDOG_AUTOMONITOR_BY_ONESHOT
+	bool "Oneshot callback"
+	depends on ONESHOT
+
 config WATCHDOG_AUTOMONITOR_BY_TIMER
 	bool "Timer callback"
+	depends on TIMER
+
+config WATCHDOG_AUTOMONITOR_BY_WDOG
+	bool "Wdog callback"
 
 config WATCHDOG_AUTOMONITOR_BY_WORKER
 	bool "Worker callback"
```

Comment:
Ditto

Reply:
And also to Idle callback, we need to improve our documentation and have more info on Kconfig makes user's life better

Reply:
Done